bundles/org.openhab.core.config.discovery.addon/src/main/java/org/openhab/core/config/discovery/addon/finders/MDNSAddonSuggestionFinder.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.config.discovery.addon.finders;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.jmdns.ServiceEvent;
+import javax.jmdns.ServiceInfo;
+import javax.jmdns.ServiceListener;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.addon.AddonDiscoveryMethod;
+import org.openhab.core.addon.AddonInfo;
+import org.openhab.core.io.transport.mdns.MDNSClient;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * This is a {@link MDNSAddonSuggestionFinder} for finding suggested Addons via MDNS.
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = AddonSuggestionFinder.class, name = MDNSAddonSuggestionFinder.SERVICE_NAME)
+public class MDNSAddonSuggestionFinder extends BaseAddonSuggestionFinder {
+
+    public static final String SERVICE_TYPE = "mdns";
+    public static final String SERVICE_NAME = SERVICE_TYPE + ADDON_SUGGESTION_FINDER;
+
+    /**
+     * Anonymous ServiceListener implementation that ignores call-backs.
+     */
+    private final ServiceListener noOp = new ServiceListener() {
```

Comment:
Why is this noOp? If there is a service added, this suggestion finder should be informed, and there should potentially be a (partial) suggestion rescan. Equally, when one is removed, corresponding suggestions should get removed. One could imagine a grace period for removal.

Reply:
Maybe also check mDNS configuration properties if some of these parameters (grace period, rescan...) exist there already.

Reply:
When testing this, I ran into the issue the suggestion list was created before the MDNS scan was finished. They never got updated, so where never available in the REST API.

Reply:
An alternative could be to rescan on a regular interval.

Reply:
Yes. To all of the above. I will adjust this, but see comment below..

Reply:
PS in this case I will ignore any discovered device removal notification, (and thus grace period etc.), because if something was once discoverable on the network, it is probably sufficient grounds for _suggesting_ to install the addon.

Reply:
> PS in this case I will ignore any discovered device removal notification, (and thus grace period etc.), because if something was once discoverable on the network, it is probably sufficient grounds for _suggesting_ to install the addon.

Works for me. A restart will remove it anyway.

Reply:
Ok. I changed it to use the mDNS push notifications, rather than the pull model. This means we could remove the scheduled `Future<>` task.